metpy/calc/indices.py

```diff
@@ -51,3 +52,51 @@ def precipitable_water(dewpt, p, top=400 * units('hPa')):
     pw = -1. * (np.trapz(w.magnitude, pres_layer.magnitude) * (w.units * pres_layer.units) /
                 (g * rho_l))
     return pw.to('millimeters')
+
+
+@exporter.export
+@check_units('[speed]', '[speed]', '[pressure]', '[length]')
+def mean_wind_pressure_weighted(u, v, p, hgt, depth, bottom=None, obs_only=False):
```

Comment:
Do you want to require the heights and depth? Get layer will work without them, so they could be defaulted to `None`.